lib/app/user.ex

```diff
@@ -0,0 +1,41 @@
+defmodule App.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias App.{User, Comment, Version}
+
+  @moduledoc """
+  user schema, define changeset to validate issue params
+  """
+
+  schema "users" do
+    field :login, :string
+    field :user_id, :integer
+    field :avatar_url, :string
+    field :html_url, :string
+
+    has_many :comments, Comment, foreign_key: :deleted_by
+    has_many :versions, Version, foreign_key: :author
+
+    timestamps()
+  end
+
+  @doc false
```

Comment:
Just a syntax preference (something I have found easier to read and maintain). Adding `@required_attrs` and `@optional_attrs` is a convention I have seen around and it is really nice to work with.

```elixir
@required_attrs ~w(user_id login avatar_url html_url)a
@optional_attrs ~w(inserted_at updated_at)a
def changeset(%User{} = user, attrs) do
  user
  |> cast(attrs, @optional_attrs ++ @required_attrs)
  |> cast_assoc(:comments, require: true)
  |> validate_required(@required_attrs)
end
```

Reply:
Also, do you plan to overwrite the `updated_at` or `inserted_at` attrs? They should be automatically handled by ecto, see here: https://github.com/elixir-ecto/ecto/blob/7ede5a2d74ef9cb239c1756d1f83cdbba4ba5d23/lib/ecto/schema.ex#L576

Reply:
For the uses I think the timestamp can be handled automatically by Ecto. We only need to overwrite the timestamps for the issue and comment schemas